code/game/gamemodes/gang/dominator.dm

```diff
@@ -48,6 +49,20 @@
 	if(gang && gang.is_dominating)
 		var/time_remaining = gang.domination_time_remaining()
 		if(time_remaining > 0)
+			var/open = 0
+			for(var/turf/T in circleviewturfs(center=src,radius=3))
+				if(!istype(T, /turf/closed))
+					open++
+			if(open < 40)
+				gang.domination_timer += 2
+				playsound(loc, 'sound/machines/buzz-two.ogg', 50, 0)
+				if(spam_prevention < 6)
+					spam_prevention++
+				else
+					gang.message_gangtools("Warning: There are too many walls around your dominator, its signal is being blocked!")
+					say("Warning: There are too many walls around your dominator, its signal is being blocked!")
```

Comment:
"your" is weird here.
I could be a sentient chicken (thus understanding common) and the dominator would be """mine"""